lib/lottery/megamillion.js

```diff
@@ -0,0 +1,131 @@
+const axios = require('axios');
+
+const options = {
+  method: 'GET',
+  url: 'https://www.megamillions.com/cmspages/utilservice.asmx/GetLatestDrawData',
+};
+
+async function makeMegamillionCall() {
+  const response = await axios.request(options);
+  return response.data;
+}
+
+function parseRawData(rawData) {
+  const start = rawData.indexOf('{');
+  const end = rawData.lastIndexOf('}');
+  if (start < 0 || end < 0) {
+    return '';
+  }
+  return rawData.substring(start, end + 1);
+}
+
+function standardizeData(rawData) {
+  const dataJson = JSON.parse(parseRawData(rawData));
+  const {
+    N1, N2, N3, N4, N5, MBall, Megaplier, PlayDate,
+  } = dataJson.Drawing;
+  const finalResult = {
+    type: 'megamillions',
+    N1,
+    N2,
+    N3,
+    N4,
+    N5,
+  };
+
+  finalResult.NS = MBall;
+  finalResult.multiplier = Number(Megaplier);
+  finalResult.date = PlayDate;
+  finalResult.jackpot = dataJson.Jackpot;
```

Comment:
Dont use playdate here, the timestamp here is the time we generate the result.
why we need jackpot?

Reply:
show how much we can earn!